src/scheduler.rs

```diff
@@ -596,20 +369,20 @@ impl RoleGroupEligibleNodes {
     fn preferred_node_or<F>(
         &self,
         pod: &PodIdentity,
-        preferred: Option<&NodeIdentity>,
+        preferred: Option<NodeIdentity>,
         default: F,
     ) -> Option<NodeIdentity>
     where
         F: Fn(&PodIdentity, &Vec<NodeIdentity>) -> Option<NodeIdentity>,
     {
         match self
             .node_set
-            .get(&pod.role)
-            .and_then(|role| role.get(&pod.group))
+            .get(&pod.role().to_string())
+            .and_then(|role| role.get(&pod.group().to_string()))
```

Comment:
I think you can just use `pod.role` and `pod_group` or with reference. The to_string() is not required.

Reply:
No because these fields are private.

Reply:
yeah i missed the brackets.